packages/library-legacy-sham/src/versioned-transaction.ts

```diff
@@ -0,0 +1,48 @@
+import {
+    BaseTransaction,
+    CompilableTransaction,
+    compileMessage,
+    createTransaction,
+    getTransactionDecoder,
+    ITransactionWithSignatures,
+} from '@solana/transactions';
+import { VersionedMessage } from '@solana/web3.js-legacy/declarations';
+
+import { createUnimplementedFunction, getUnimplementedError } from './unimplemented';
+
+export class VersionedTransaction {
+    #tx: BaseTransaction | (BaseTransaction & ITransactionWithSignatures);
+    constructor(message: VersionedMessage, _signatures?: Array<Uint8Array>) {
+        if (message) {
+            throw getUnimplementedError('Constructing a `VersionedTransaction`');
+        }
+        this.#tx = createTransaction({ version: 0 });
+    }
+    addSignature = createUnimplementedFunction('VersionedTransaction#addSignature');
+    serialize = createUnimplementedFunction('VersionedTransaction#serialize');
+    sign = createUnimplementedFunction('VersionedTransaction#sign');
+    get message(): VersionedMessage {
+        throw getUnimplementedError('VersionedTransaction#message (getter)');
+    }
+    get signatures(): Uint8Array[] {
+        const {
+            header: { numSignerAccounts },
+            staticAccounts,
+        } = compileMessage(this.#tx as unknown as CompilableTransaction);
+        return staticAccounts.slice(0, numSignerAccounts).reduce((acc, account) => {
+            acc.push('signatures' in this.#tx ? this.#tx.signatures[account] : new Uint8Array(64));
+            return acc;
+        }, [] as Uint8Array[]);
+    }
+    get version() {
+        return this.#tx.version;
+    }
+    static deserialize(serializedTransaction: Uint8Array): VersionedTransaction {
```

Comment:
This PR mostly exists so that you can do `VersionedTransaction.deserialize` followed by `transaction.signatures`.